cli/cmd/install.go

```diff
@@ -205,6 +206,41 @@ func checkNoConfig(ctx context.Context, k8sAPI *k8s.KubernetesAPI) error {
 	return nil
 }
 
+type GatewayAPICRDs int
+
+const (
+	Absent GatewayAPICRDs = iota
+	Linkerd
+	External
+)
+
+// checkGatewayAPICRDs returns true if the Gateway API CRDs are installed in the
+// cluster, and false otherwise.
+func checkGatewayAPICRDs(ctx context.Context, k8sAPI *k8s.KubernetesAPI) (GatewayAPICRDs, error) {
+	crds := k8sAPI.Apiextensions.ApiextensionsV1().CustomResourceDefinitions()
+	result := Absent
+	names := []string{
+		"httproutes.gateway.networking.k8s.io",
+		"grpcroutes.gateway.networking.k8s.io",
+		"tlsroutes.gateway.networking.k8s.io",
+		"tcproutes.gateway.networking.k8s.io",
```

Comment:
I think we only want to *require* v1 CRDs, which at this point ecludes tlsroute and tcproute.

Reply:
As written, if we see ANY of the gateway API CRDs, we treat the mode as External.  But we could e.g. only check for httproutes or grpcroutes.